src/modules/account/account.service.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { CreateAccountDto } from './dtos/create-account.dto';
+import { generate } from 'generate-password';
+import { UpdateAccountDto } from './dtos/update-account.dto';
+import { AccountDAL } from './dal/account.dal';
+import { validateNonEmptyObject } from 'src/common/util/validator';
+
+@Injectable()
+export class AccountService {
+  constructor(private accountDAL: AccountDAL) {}
+
+  async createAccount(data: CreateAccountDto, userId: number) {
+    if (!userId) throw new BadRequestException('User ID is required.');
+
+    const user = await this.accountDAL.findUserById(userId);
```

Comment:
i have created a function in my auth controller to get the user or throw error , you can use that